src/core/ext/filters/client_channel/resolver/dns/c_ares/dns_resolver_ares.cc

```diff
@@ -358,74 +363,86 @@ class AresClientChannelDNSResolverFactory : public ResolverFactory {
 
 class AresDNSResolver : public DNSResolver {
  public:
-  class AresRequest : public DNSResolver::Request {
+  class AresRequest {
    public:
     AresRequest(
         absl::string_view name, absl::string_view default_port,
         grpc_pollset_set* interested_parties,
         std::function<void(absl::StatusOr<std::vector<grpc_resolved_address>>)>
-            on_resolve_address_done)
+            on_resolve_address_done,
+        AresDNSResolver* resolver, intptr_t aba_token)
         : name_(std::string(name)),
           default_port_(std::string(default_port)),
           interested_parties_(interested_parties),
-          on_resolve_address_done_(std::move(on_resolve_address_done)) {
+          on_resolve_address_done_(std::move(on_resolve_address_done)),
+          completed_(false),
+          resolver_(resolver),
+          aba_token_(aba_token) {
       GRPC_CARES_TRACE_LOG("AresRequest:%p ctor", this);
       GRPC_CLOSURE_INIT(&on_dns_lookup_done_, OnDnsLookupDone, this,
                         grpc_schedule_on_exec_ctx);
+      MutexLock lock(&mu_);
+      pollset_set_ = grpc_pollset_set_create();
+      grpc_pollset_set_add_pollset_set(interested_parties, pollset_set_);
```

Comment:
Is this right - should this not be `grpc_pollset_set_add_pollset_set(pollset_set_, interested_parties);` ?

We want fds added to `pollset_set_` to propagate into the callers's `interested_parties_`, which I think we need to change this in order to do?

Otherwise, this might only be working because of backup pollers. E.g. if we do a DNS resolution manually, I suspect it will be slow.

Reply:
Indeed. I dialed the backup poller back to 10s intervals and tried external resolution ... it took 10s. Great catch. Fixed.